projects/igniteui-angular/.eslintrc.json

```diff
@@ -41,7 +41,14 @@
         ],
         "id-blacklist": "off",
         "id-match": "off",
-        "no-underscore-dangle": "off"
+        "no-underscore-dangle": "off",
+        "no-restricted-syntax": [
+          "error",
+          {
+            "selector": "Decorator[expression.callee.name=HostBinding] Literal[value='attr.class']",
+            "message": "Do not use `attr.class` HostBinding in public-facing components"
+          }
+        ]
```

Comment:
Threw in a lint rule for good measure. Might be a bit overkill, but really don't want to leave it to chance this is never used since Angular's test won't catch it. Adapted 
Also it was quite fun, [`no-restricted-syntax`](https://eslint.org/docs/latest/rules/no-restricted-syntax) with AST selectors seems to be pretty much all-powerful 😁
Playground link for the curious: [https://typescript-eslint.io/play/#...](https://typescript-eslint.io/play/#ts=4.5.2&sourceType=module&showAST=es&code=PQKgsAUABFCmDOAbAlgOwC5VQewLQCcF19kBjdWAE13gE8MBDADwC4oBtSGbqAclnz5s+XgBouPKAG8JkqACJ4sRLHLD5beQBFVwhumHtYTAA6F48ZNlQA6Ug0QrYN1AwC2sALwAJbPHQAQmiUaADmALpQADLIFPgO7ABuDgCuXrz6xHaIDBa84fLi0HIKHhYMobAaCgCMUFrYWNiYKUpQAAaZ+Nm58O1Qvv5BqCGooVBoUCYpAEYopLgAZgykYVCk2G4m1rAY8PKy3AC+RXIyxXKKyqoG+NXauvG3RqbmltZ2Dk4u7l6DgcEwuEXmYEO9bAx8KEUh49jYAAw2ZKINKeDLoLKkHJ5AqnEryMrwCpVTQAJnqjRwLTanQx3SxvX6-2Go3Gk2mczISxWaw2Wx2ewOF2OEnCkBAwEgkEoqhyhHW1n8Az8AJGYQA3FKIKBwNAQFAAAIAC2QlBlqC4EsgDIsUAoSvOMANzMBYwAFOjMdj4LwAJQSDnzKCVTBGlUAYW9bt9bH8JDG0kOhHQKXwqD4vE1xSOkCOQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3QAacDUjoAHoizJ46IngCGqACL9YmNXl3Cwo9FIC+IM0A)